api/http/handler/endpointedge/async.go

```diff
@@ -0,0 +1,312 @@
+
+// @id endpointAsync
+// @summary Get environment(endpoint) status
+// @description Environment(Endpoint) for edge agent to check status of environment(endpoint)
+// @description **Access policy**: restricted only to Edge environments(endpoints) TODO: with mTLS cert....
+// @tags endpoints
+// @security ApiKeyAuth
+// @security jwt
+// @param id path int true "Environment(Endpoint) identifier"
+// @success 200 {object} AsyncResponse "Success"
+// @failure 400 "Invalid request"
+// @failure 403 "Permission denied to access environment(endpoint)"
+// @failure 404 "Environment(Endpoint) not found"
+// @failure 500 "Server error"
+// @router /endpoints/edge/async/ [post]
+func (handler *Handler) endpointAsync(w http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	// TODO: get endpointID from the mTLS cert info
+	edgeIdentifier := r.Header.Get(portainer.PortainerAgentEdgeIDHeader)
+	if edgeIdentifier == "" {
+		logrus.WithField("portainer.PortainerAgentEdgeIDHeader", edgeIdentifier).Debug("missing agent edge id")
+
+		return &httperror.HandlerError{http.StatusInternalServerError, "missing Edge identifier", errors.New("missing Edge identifier")}
+	}
+
+	// TODO: if the mTLS certs are valid, and we don't have a matching environment registered, CREATE IT (and maybe untrusted...)
+	endpoint, err := handler.getEdgeEndpoint(edgeIdentifier)
+	if err != nil {
+		// TODO: if its a valid cert, or the user hasn't limited to mTLS / portainer set id, the
+		// create new untrusted environment
+		// portainer.HTTPResponseAgentPlatform tells us what platform it is too...
+		logrus.WithField("portainer.PortainerAgentEdgeIDHeader", edgeIdentifier).Debug("edge id not found in existing endpoints!")
+	}
+	// if agent mTLS is on, drop the connection if the client cert isn't CA'd (or if its revoked)
+	err = handler.requestBouncer.AuthorizedEdgeEndpointOperation(r, endpoint)
+	if err != nil {
+		return &httperror.HandlerError{http.StatusForbidden, "Permission denied to access environment", err}
+	}
+
+	// Any request we can identify as coming from a valid agent is treated as a Ping
+	endpoint.LastCheckInDate = time.Now().Unix()
+	endpoint.Status = portainer.EndpointStatusUp
+	err = handler.DataStore.Endpoint().UpdateEndpoint(endpoint.ID, endpoint)
+	if err != nil {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to Unable to persist environment changes inside the database", err}
+	}
+
+	// TODO: update endpoint contact time
+
+	var payload AsyncRequest
+	err = request.DecodeAndValidateJSONPayload(r, &payload)
+	if err != nil {
+		// an "" request ~~ same as {}
+		logrus.WithError(err).WithField("payload", r).Debug("decode payload")
+	}
+
+	asyncResponse := AsyncResponse{
+		ServerCommandId: "8888", // the most current id of a new command on the server
+	}
+
+	// TODO: need a way to detect that these are changed, and send them to the agent...
+	// CommandInterval time.Duration `json: optional`
```

Comment:
are these meant to be saved as settings in the datastore? maybe allow the user to change it through the UI?